ktor-server/ktor-server-cio/jvm/test/io/ktor/tests/server/cio/TestConnectorInterfaceUsed.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.tests.server.cio
+
+import io.ktor.server.cio.*
+import io.ktor.server.engine.*
+import kotlin.test.*
+
+class TestConnectorInterfaceUsed {
+    @Test
+    fun testConnectorListenHost() {
+        val engine = CIOApplicationEngine(
+            applicationEngineEnvironment {
+                connector {
+                    host = "some/illegal/host/name"
+                    port = 9091
+                }
+            }
+        ) {}
+
+        assertFails {
+            engine.start()
+
+            // this shouldn't happen
```

Comment:
maybe add explicit `fail`?

Reply:
We should invoke stop anyway while assertFails will do fail